config/src/main/java/com/typesafe/config/impl/ConfigParser.java

```diff
@@ -192,10 +192,39 @@ private void parseInclude(Map<String, AbstractConfigValue> values, ConfigNodeInc
             }
         }
 
+        private ConfigConditional parseConditional(ConfigNodeConditional n) {
+            SubstitutionExpression left = null;
+            AbstractConfigValue right = null;
+
+            AbstractConfigObject body = parseObject((ConfigNodeObject) n.body());
+
+            for(AbstractConfigNode child: n.children()) {
+                if (child instanceof ConfigNodeSingleToken) {
+                    Token token = ((ConfigNodeSingleToken) child).token();
+
+                    if (Tokens.isSubstitution(token)) {
+                        List<Token> pathTokens = Tokens.getSubstitutionPathExpression(token);
+                        Path path = PathParser.parsePathExpression(pathTokens.iterator(), token.origin());
+                        left = new SubstitutionExpression(path, false);
+                    }
+                } else if (child instanceof AbstractConfigNodeValue) {
+                    right = parseValue((AbstractConfigNodeValue)child, null);
+                }
+            }
+
+            if (left == null)
+                throw new ConfigException.BugOrBroken("Conditional expression did not have a left hand substitution");
```

Comment:
Yea it is.  A common pattern in this lib is to check for various badness at tokenization time and again at parse time.  Presumably in case one piece changes independently of the other.